### ISSUE TITLE
autoTimestamp should support jsr310 dates out-of-the-box

### ISSUE THREAD

**Issue by allnightlong** [grails/grails-data-mapping#780]
Currently for domain like:
```
class Book {
	String title
	String author
	LocalDateTime dateCreated
}
```
saving
`new Book(title: 'title', author: 'author').save(flush: true)`
throws exception:
```
Could not find matching constructor for: java.time.LocalDateTime(java.lang.Long). Stacktrace follows:
java.lang.reflect.InvocationTargetException: null
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by graemerocher** [comment]
Relates to #704

**Comment by jameskleeh** [comment]
Fixed in 6.0.1
